source/common/upstream/cluster_manager_impl.cc

```diff
@@ -2399,6 +2399,20 @@ Http::ConnectionPool::InstancePtr ProdClusterManagerFactory::allocateConnPool(
   }
   if (protocols.size() == 1 && protocols[0] == Http::Protocol::Http2 &&
       context_.runtime().snapshot().featureEnabled("upstream.use_http2", 100)) {
+    if (host->cluster().clusterType().has_value() &&
+        host->cluster().clusterType()->name() == "envoy.clusters.reverse_connection") {
+      ENVOY_LOG_MISC(debug, "allocating reverse connection pool");
+      auto* factory = Config::Utility::getFactoryByName<Http::Http2::ReverseConnPoolFactory>(
+          "envoy.http.reverse_conn.default");
```

Comment:
It is looking for the factory by name, and this will limit this code path to a specific cluster with this name, could you use the `Factory* getFactory(const ProtoMessage& message)` by type? And then this lookup will be extendable. See above comments for the `ReverseConnPoolFactory`

Reply:
Okay, let me try to convert it to the `TypedFactory` as suggested above and switch the lookup.